pkg/operation/botanist/addons.go

```diff
@@ -334,14 +334,19 @@ func (b *Botanist) generateCoreAddonsChart(ctx context.Context) (*chartrenderer.
 		return nil, err
 	}
 
+	clusterCASecret, found := b.SecretsManager.Get(v1beta1constants.SecretNameCACluster)
+	if !found {
+		return nil, fmt.Errorf("secret %q not found", v1beta1constants.SecretNameCACluster)
+	}
+
 	apiserverProxyConfig := map[string]interface{}{
 		"advertiseIPAddress": b.APIServerClusterIP,
 		"proxySeedServer": map[string]interface{}{
 			"host": kasFQDN,
 			"port": "8443",
 		},
 		"webhook": map[string]interface{}{
-			"caBundle": b.LoadSecret(v1beta1constants.SecretNameCACluster).Data[secrets.DataKeyCertificateCA],
+			"caBundle": clusterCASecret.Data[secrets.DataKeyCertificateBundle],
```

Comment:
BTW: `apiserver-proxy-pod-mutator` should probably get its own server cert instead of serving with `kube-apiserver`'s server cert.

Reply:
Not sure, definitely not in this PR 😄